api/src/main/java/org/apache/iceberg/util/CharSequenceSet.java

```diff
@@ -172,7 +172,7 @@ public boolean equals(Object o) {
 
   @Override
   public int hashCode() {
-    return Objects.hash(wrapperSet);
+    return Objects.hashCode(wrapperSet);
```

Comment:
I'm not very comfortable changing this because it's a public api. While I don't think we actually serialize this anywhere, if someone did this would probably cause an issue. That said if as a community we feel confident this isn't going to be breaking we can change it.

Objects.hash will hash the hash of wrapperSet 
Objects.hashCode will hash wrapperSet

Reply:
revert